src/libertem/io/dataset/frms6.py

```diff
@@ -153,6 +170,28 @@ def _read_file_header(path):
     return header
 
 
+def _get_sig_shape(path, bin_factor):
+    path, ext = os.path.splitext(path)
+    if ext == ".hdr":
+        pattern = "%s_*.frms6" % path
+    elif ext == ".frms6":
+        pattern = "%s*.frms6" % (
+            re.sub(r'[0-9]+$', '', path)
+        )
+    else:
+        raise DataSetException("unknown extension: %s" % ext)
```

Comment:
This is duplicated with the `_pattern` function, maybe `_pattern` can be refactored to use the same code as this?